samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/export/ExportMarkupTextServlet.java

```diff
@@ -0,0 +1,178 @@
+				String points = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages", "points_lower_case");
+				String discount = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.SamLite", "samlite_discount");
+				String randomize = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.SamLite", "example_mc_question_random");
+				String rationale = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.SamLite", "example_mc_question_rationale");
+				String str_true = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.SamLite", "samlite_true");
+				String str_false = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.SamLite", "samlite_false");
+				bundle.put("points", points);
+				bundle.put("discount", discount.toLowerCase());
+				bundle.put("randomize", randomize);
+				bundle.put("rationale", rationale);
+				bundle.put("true", str_true);
+				bundle.put("false", str_false);
+				
+				String markupText = assessmentService.exportAssessmentToMarkupText(assessment, bundle);
+				 
+				res.setContentType("text/plain");
+				String filename = "exportAssessment.txt";
+				res.setHeader("Content-Disposition", "attachment;filename=\""
+						+ filename + "\";");
+
+				res.setContentLength(markupText.length());
+				PrintWriter out = res.getWriter();
+				out.println(markupText);
+				out.close();
+				out.flush();
+			}
+		}else{
+			if(success == AssessmentService.UPDATE_ERROR_DRAW_SIZE_TOO_LARGE){  		    		
+				String err=ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages","update_pool_error_size_too_large");
+				req.setAttribute("error", err);
+			}else{
+				String err=ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages","update_pool_error_unknown");
+				req.setAttribute("error", err);
+			}
+
+			String path = "/jsf/qti/poolUpdateError.faces";
+			RequestDispatcher dispatcher = req.getRequestDispatcher(path);
+			dispatcher.forward(req, res);
+		}
+	}
+
+	private boolean isOwner(String agentId, String ownerId) {
+		boolean isOwner = false;
+		isOwner = agentId.equals(ownerId);
+		return isOwner;
+	}
+
+	private String getAgentString(HttpServletRequest req,
+			HttpServletResponse res) {
+		String agentIdString = AgentFacade.getAgentString();
+		if (agentIdString == null || agentIdString.equals("")) { // try this
+			PersonBean person = (PersonBean) ContextUtil
+					.lookupBeanFromExternalServlet("person", req, res);
+			agentIdString = person.getAnonymousId();
+		}
+		return agentIdString;
+	}
+
+	public boolean canExport(HttpServletRequest req, HttpServletResponse res,
+			String agentId, String currentSiteId, String createdBy) {
+		log.debug("agentId={}" + agentId);
+		log.debug("currentSiteId=" + currentSiteId);
+		boolean hasPrivilege_any = hasPrivilege(req, "assessment.editAssessment.any",

```

Comment:
I wouldn't mind seeing some constants for these permission strings. Maybe also for the message bundle key strings up above.